[LI-HOTFIX] add prepare-commit-msg hook to help format commit message

xiowu0 opened the pull request:
TICKET =
LI_DESCRIPTION =
Automatically reformat commit messages:
1) extract ticket number and reformat for cherry-picks
2) provide a standard template for Hotfix patches.

Usage:
cherry-pick (please use -x):
git cherry-pick -e <hash> -x

Hotfix:
git commit ...
EXIT_CRITERIA = MANUAL [""]

xiowu0 committed:
[LI-HOTFIX] add prepare-commit-msg hook to help format commit message
TICKET =
LI_DESCRIPTION =
Automatically reformat commit messages:
1) extract ticket number and reformat for cherry-picks
2) provide a standard template for Hotfix patches.

Usage:
cherry-pick (please use -x):
git cherry-pick -e <hash> -x

Hotfix:
git commit ...
EXIT_CRITERIA = MANUAL [""]

jonlee2 commented on .githooks/prepare-commit-msg:
```diff
@@ -0,0 +1,140 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage:
```
Do we need to install this hook explicitly or will this be picked up automatically?

Also, can you add a short description on what this hook does?

xiowu0 replied:
We need to install the hook explicitly (via provided setup.sh).  updated description and added setup instruction to the commit message.

xiowu0 committed:
[LI-HOTFIX] add prepare-commit-msg hook to help format commit message
TICKET =
LI_DESCRIPTION =
Automatically reformat commit messages via git hook:
1) extract ticket number and reformat for cherry-picks
2) provide a standard template for Hotfix patches.

Setup/Install the hook:
.githooks/setup.sh
Remove the hook:
.githooks/uninstall.sh

Once installed, the hook will be automatically invoked through the following usages:
  cherry-pick (please use -x):
  git cherry-pick -e <hash> -x

  Hotfix:
  git commit ...
EXIT_CRITERIA = MANUAL [""]

jonlee2 commented on .githooks/prepare-commit-msg:
```diff
@@ -0,0 +1,147 @@
+	else:
+		actual_msg.append(line)
+
+# skip already formatted commits
```
Nit: why not moving this to line 74, right after you get firstline?

xiowu0 committed:
[LI-HOTFIX] add prepare-commit-msg hook to help format commit message
TICKET =
LI_DESCRIPTION =
Automatically reformat commit messages via git hook:
  1) extract ticket number and reformat for cherry-picks
  2) provide a standard template for Hotfix patches.

Setup/Install the hook:
  .githooks/setup.sh
Remove the hook:
  .githooks/uninstall.sh

Once installed, the hook will be automatically invoked through the following usages:
  cherry-pick (please use -x):
    git cherry-pick -e <hash> -x

  Hotfix:
    git commit ...
EXIT_CRITERIA = MANUAL [""]
